framework/src/problems/EigenProblem.C

```diff
@@ -371,28 +371,10 @@ EigenProblem::initEigenvector(const Real initial_value)
 
   std::vector<VariableName> var_names = getVariableNames();
 
-  // Count how many dofs we have
-  for (auto & vn : var_names)
-  {
-    MooseVariableBase * var = nullptr;
-    if (hasScalarVariable(vn))
-      var = &getScalarVariable(0, vn);
-    else
-      var = &getVariable(0, vn);
-    if (var->eigen())
-    {
-      std::set<dof_id_type> var_indices;
-      for (unsigned int vc = 0; vc < var->count(); ++vc)
-      {
-        var_indices.clear();
-        _nl_eigen->system().local_dof_indices(var->number() + vc, var_indices);
-        n_dofs += var_indices.size();
-      }
-    }
-  }
-
-  // Note that even n_dofs is equal to zero, we can not return here because we might
-  // hit MPI deadlock when 'close' is called at the end of this function
+  // In multiphysics simulations, n_dofs here may not be the number of eigen dofs,
+  // but it is fine since we need to only scale eigenvector so that the norm of
+  // initial vector does not depend on problem size
+  n_dofs = _nl_eigen->solution().size();
```

Comment:
we possibly do not need the check on line 388. Also since we are not using `n_dofs`, we possibly can remove this line completely unless you want to set the value to `initial_value / n_dofs` later.

Reply:
Removed